app/controllers/api/v1/planning_applications_controller.rb

```diff
@@ -26,10 +26,15 @@ def decision_notice
   end
 
   def create
-    site_id = create_site
     @planning_application = PlanningApplication.new(
-      full_planning_params.merge!(site_id: site_id, local_authority_id: @current_local_authority.id),
+      full_planning_params.merge!(
+        local_authority_id: @current_local_authority.id,
+      ),
     )
+    if site_params
+      @planning_application.update!(site_params)
```

Comment:
You're calling update!, which saves the record, and raises an error if not valid. Ideally we would only call save on this record on line 38.

I would propose merging the site_params into full_planning_params in the method on line 111

Reply:
this site params separate call was to make sure the application didn't error when the site_params are nil - which happens in a few of the tests. I'll change update! to update so it doesn't save the record, I did think maybe there was a better way to do this (maybe with a @planning_application << method) but it can't be merged into the one above or a bunch of tests fail..

Reply:
I've solved this by using assign_attributes instead of update - we still need the "if" check provided by the site params method hence the need to keep it separate - but if you know of a way to keep the if check and merge that into the other method in an elegant way I'm happy to do it, just don't see how its possible

Reply:
Using `assign_attributes` fixes the "double save" and raising an application error issue, so the code now works as we want :-)

My following suggestion is only a different way to achieve the same thing, but I think is a little bit simpler?

If you replace `full_planning_params` with the following:

```ruby
def full_planning_params
  planning_application_params.merge!({
    proposal_details: (params[:proposal_details].to_json if params[:proposal_details].present?),
    constraints: (params[:constraints].to_json if params[:constraints].present?),
    audit_log: params.to_json,
    uprn: params[:site][:uprn],
    address_1: params[:site][:address_1],
    address_2: params[:site][:address_2],
    town: params[:site][:town],
    postcode: params[:site][:postcode],
  })
end
```

Then I think you can remove the `site_params` method, and lines 34-36. I think that's a bit cleaner?